### PR TITLE
CF_GMSA_OU environment variable

### PR DESCRIPTION
**Pull request by gustavo-armenta**
*Issue #, if available:*

*Description of changes:*
CF_GMSA_OU environment variable to support arbitrary GMSA distinguished name

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

**Comment by jamolina** [comment]
Thank you for your contribution @gustavo-armenta  . We'll review it and let you know if we have any questions

**Review by smhmhmd — APPROVED** [review]
Thanks Gustavo

File: auth/kerberos/src/krb.cpp [hunk 1]
```diff
@@ -661,9 +662,13 @@ std::pair<int, std::string> get_gmsa_krb_ticket( std::string domain_name,
      *   Accounts,DC=contoso,DC=com' -s sub  "(objectClass=msDs-GroupManagedServiceAccount)"
      *   msDS-ManagedPassword
      */
+    std::string gmsa_ou = std::string( ",CN=Managed Service Accounts," );
+    if ( getenv(ENV_CF_GMSA_OU) != NULL)
+    {
+        gmsa_ou = std::string( "," ) + getenv(ENV_CF_GMSA_OU)+ std::string( "," );
```
Comment (smhmhmd) [inline auth/kerberos/src/krb.cpp]: Would be good to use`std::string(getenv(ENV_CF_GMSA_OU))`, since getenv returns char *

**Review by saikiranakula-amzn — APPROVED** [review]

File: auth/kerberos/src/krb.cpp (hunk 1 above)
Comment (saikiranakula-amzn) [reply auth/kerberos/src/krb.cpp]: new cod push addressed the comment

**Pull request merged**